File: src/db/model/kline.py
from sqlalchemy import Integer, Column, DateTime, Float, String

from db.model.base import Base


class Kline(Base):
    __tablename__ = 'klines'
    id = Column(Integer, primary_key=True)
    symbol = Column(String)
    interval = Column(String)  # 1m, 5m, 15m, 1h, 1d
    timestamp = Column(DateTime)
    open = Column(Float)
    high = Column(Float)
    low = Column(Float)
    close = Column(Float)
    volume = Column(Float)
    close_time = Column(DateTime)
    quote_asset_volume = Column(Float)
    number_of_trades = Column(Integer)
    taker_buy_base_asset_volume = Column(Float)
    taker_buy_quote_asset_volume = Column(Float)
    ignore = Column(String)

File: binance-data-collection/src/main/java/com/oyakov/binance_data_collection/model/binance/Kline.java
package com.oyakov.binance_data_collection.model.binance;

import com.fasterxml.jackson.annotation.JsonAlias;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;

@Data
@AllArgsConstructor
@NoArgsConstructor
@JsonIgnoreProperties(ignoreUnknown = true)
@JsonInclude(JsonInclude.Include.NON_NULL)
public class Kline {
    @JsonAlias("s")
    private String symbol;
    @JsonAlias("i")
    private String interval;
    @JsonAlias("t")
    private long openTime;
    @JsonAlias("o")
    private BigDecimal open;
    @JsonAlias("h")
    private BigDecimal high;
    @JsonAlias("l")
    private BigDecimal low;
    @JsonAlias("c")
    private BigDecimal close;
    @JsonAlias("v")
    private BigDecimal volume;
    @JsonAlias("T")
    private long closeTime;
    @JsonAlias("x")
    private boolean closed;
}
